crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -76,17 +78,79 @@ impl DynamicScene {
         scene
     }
 
+    /// Create a new dynamic scene from a given world at a given root.
+    pub fn from_world_at_root(
+        world: &World,
+        root: Entity,
+        type_registry: &AppTypeRegistry,
+    ) -> Self {
+        let mut scene = DynamicScene::default();
+        let type_registry = type_registry.read();
+
+        // Extract all entities that are a descendent of root
+        let mut entities_of_interest = Vec::new();
+        let mut entities_to_process = vec![root];
+        while let Some(entity) = entities_to_process.pop() {
+            entities_of_interest.push(entity);
+            if let Some(children) = world.entity(entity).get::<Children>() {
+                entities_to_process.extend(children.iter());
+            }
+        }
+
+        for archetype in world.archetypes().iter() {
```

Comment:
Is there no way to look up which archetype an entity is in by its Entity?

Reply:
changed the code to iterate over entities and get their archetypes instead of iterating over archetypes and checking for entities